it/uk/gov/hmrc/minorentityidentificationfrontend/views/uaViews/CaptureOfficePostcodeViewTests.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.views.uaViews
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.{Document, Element}
+import org.jsoup.select.Elements
+import play.api.libs.ws.WSResponse
+import uk.gov.hmrc.minorentityidentificationfrontend.assets.MessageLookup.{Base, BetaBanner, Header, CaptureOfficePostcode => messages}
+import uk.gov.hmrc.minorentityidentificationfrontend.assets.TestConstants.testSignOutUrl
+import uk.gov.hmrc.minorentityidentificationfrontend.config.AppConfig
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.ComponentSpecHelper
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.ViewSpecHelper.ElementExtensions
+
+trait CaptureOfficePostcodeViewTests {
+  this: ComponentSpecHelper =>
+
+  def testCaptureOfficePostcodeView(result: => WSResponse): Unit = {
+    lazy val doc: Document = Jsoup.parse(result.body)
+    lazy val config = app.injector.instanceOf[AppConfig]
+
+
+    "have a sign out link in the header" in {
+      doc.getSignOutText mustBe Header.signOut
+    }
+
+    "have sign out link redirecting to signOutUrl from journey config" in {
+      doc.getSignOutLink mustBe testSignOutUrl
+    }
+
+    "have the correct beta banner" in {
+      doc.getBanner.text mustBe BetaBanner.title
+    }
+
+    "have a banner link that redirects to beta feedback" in {
+      doc.getElementsByClass("govuk-link").get(1).attr("href") mustBe config.betaFeedbackUrl("vrs")
+    }
+
+    "have the correct title" in {
+      doc.title mustBe messages.title
+    }
+
+    "have the correct page header" in {
+      val headers: Elements = doc.getElementsByTag("h1")
+
+      headers.size mustBe >= (1)
+
+      headers.first.text mustBe messages.title
+    }
+
+    "have an input text box with the identifier 'officePostcode'" in {
+
+      val optInput: Option[Element] = Option(doc.getElementById("officePostcode"))
+
+      optInput match {
+        case Some(input) => input.attr("type") mustBe "text"
+        case None => fail("""Input element "chrn" cannot be found""")
```

Comment:
It should be "officePostcode" rather than "chrn"

Reply:
done